main/src/library/Applicative.flix

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright 2021 Felix Wiemuth
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// A type class for functors that support application, i.e. allow to:
+/// - Make an applicative value out of a normal value (embed it into a default context), e.g. embed `5` into `Some(5)`.
+/// - Apply a function-type applicative to a matching argument-type applicative, resulting in an applicative of
+/// the function's result type. The meaning of this application is defined by the respective instance.
+/// Conceptually this can be understood as applying functions "contained" in the first applicative to arguments
+/// in the second applicative, where the possible quantity of functions/arguments depends on the type `m`.
+/// For example, an `Option[a -> b]` can be `None`, or contain a function of type `a -> b`, and only in the
+/// latter case a function is applied. A `List[a -> b]` contains a list of functions, which are all to be
+/// applied to all arguments in the arguments list.
```

Comment:
I think one or two lines breaks in the above would be useful.

Reply:
Do you mean more paragraphs or shorter lines in general?

Reply:
Text is fine, I mean some literal line breaks, \n

Reply:
Yes, but to make more paragraphs out of the one paragraph or to have shorter lines?

Reply:
Breathing space:

```
/// A type class for functors that support application, i.e. allow to:
///
/// - Make an applicative value out of a normal value (embed it into a default context), e.g. embed `5` into `Some(5)`.
/// - Apply a function-type applicative to a matching argument-type applicative, resulting in an applicative of
///
/// the function's result type. The meaning of this application is defined by the respective instance.
```